### ISSUE TITLE
autocomplete

### ISSUE THREAD

**Issue by jonsterling** [david-christiansen/jonprl-mode#13]
When https://github.com/jonsterling/JonPRL/pull/102 is merged into JonPRL, there will be a `--list-operators` command which spits out a bunch of lines like `op-name (1;2;0)` etc, which give the names of defined operators and their arities (lists of valences).

It would be super sweet if I could type an operator name and hit tab or something, and then have the full binding structure of the term filled out with little holes that I could then edit. Does this sound feasible?

**Comment by david-christiansen** [comment]
Yes, this can be done by dynamically computing yasnippet templates. Idris-mode used to do that, but it got flaky when combined with the compiler-driven codegen. I imagine that, here, it wouldn't have that issue, at least not until JonPRL gets sweet interactive editing from the compiler's side.

**Comment by david-christiansen** [comment]
This is done now.
